src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -10081,12 +9960,11 @@ private StatementSyntax ParseLocalDeclarationStatement(SyntaxList<AttributeListS
                     awaitKeyword,
                     usingKeyword,
                     mods.ToList(),
-                    _syntaxFactory.VariableDeclaration(type, variables),
+                    _syntaxFactory.VariableDeclaration(type, _pool.ToListAndFree(variables)),
                     semicolon);
             }
             finally
             {
-                _pool.Free(variables);
                 _pool.Free(mods);
```

Comment:
currently, there's no helper for SyntaxTokenLists, but i want to add one to clean this up as well.